src/helpers/strings.ts

```diff
@@ -130,3 +179,112 @@ export function formatCurrency(
 
   return `${currencySymbol}${formattedWhole}.${formattedFraction}`;
 }
+
+export function formatFractionWorklet(fraction: string): string {
+  'worklet';
+  let leadingZeros = 0;
+  for (let i = 0; i < fraction.length; i++) {
+    if (fraction[i] === '0') {
+      leadingZeros++;
+    } else {
+      break;
+    }
+  }
+
+  const significantPart = fraction.slice(leadingZeros);
+  if (significantPart.length === 0 || /^[0]+$/.test(significantPart)) {
+    return '00';
+  }
+
+  const significantDigits = significantPart.slice(0, 2).padEnd(2, '0');
+
+  if (leadingZeros >= 4) {
+    const leadingZerosStr = leadingZeros.toString();
+    let subscriptZeros = '';
+    for (let i = 0; i < leadingZerosStr.length; i++) {
+      subscriptZeros += toSubscript(String(leadingZerosStr[i]));
+    }
+    return `0${subscriptZeros}${significantDigits}`;
+  }
+  return `${'0'.repeat(leadingZeros)}${significantDigits}`;
+}
+
+const SUBSCRIPT_THRESHOLD_MAGNITUDE = -4;
+
+export function toCompactNotation({
+  value,
+  prefix,
+  decimalPlaces,
+}: {
+  value: string | number;
+  prefix?: string;
+  decimalPlaces?: number;
+}): string {
+  'worklet';
+  const valueString = safeToStringWorklet(value);
+  if (!safeIsNumberStringWorklet(valueString)) return valueString ?? '';
+
+  const numericString = valueString;
+  const isNegative = safeLessThanWorklet(numericString, '0');
+  const absNumericString = isNegative ? safeMulWorklet(numericString, '-1') : numericString;
+  const sign = isNegative ? '-' : '';
+
+  if (safeEqualWorklet(absNumericString, '0')) {
+    const formattedValue = safeToFixedWorklet('0', decimalPlaces ?? 2);
+    return prefix ? `${prefix}${formattedValue}` : `${formattedValue}`;
+  }
+
+  const magnitude = safeOrderOfMagnitudeWorklet(absNumericString);
+
+  // Handle >= 1
+  if (safeGreaterThanOrEqualToWorklet(absNumericString, '1')) {
+    // Scientific notation for >= 1,000,000
+    if (magnitude >= 6) {
+      const divisor = safePowWorklet('10', magnitude.toString());
+      const mantissa = safeDivWorklet(absNumericString, divisor);
+      const formattedMantissa = safeToFixedWorklet(mantissa, 2);
+      const superscriptExponent = `${magnitude
+        .toString()
+        .split('')
+        .map(char => toSuperscript(char))
+        .join('')}`;
+      return prefix
+        ? `${prefix}${sign}${formattedMantissa}×10${superscriptExponent}`
+        : `${sign}${formattedMantissa}×10${superscriptExponent}`;
+    }
+
+    // Fixed decimal for 1 <= abs < 1,000,000
+    const formattedValue = safeToFixedWorklet(absNumericString, decimalPlaces ?? 2);
+    return prefix ? `${prefix}${sign}${formattedValue}` : `${sign}${formattedValue}`;
+  }
+
+  if (magnitude >= SUBSCRIPT_THRESHOLD_MAGNITUDE) {
+    const targetDecimalPlaces = decimalPlaces ?? 2;
+    let formattedValue = safeToFixedWorklet(absNumericString, targetDecimalPlaces);
+
+    const isZeroFormatted = /^[-+]?0+(\.0+)?$/.test(formattedValue);
```

Comment:
regex's should usually be defined once outside of the function because otherwise they get compiled on every call. This isn't an issue if the function is just called a couple times, but given this is used in large lists and when scrubbing the chart it's worth it here.